pkg/reconciler/tenancy/workspace/workspace_reconcile_phase.go

```diff
@@ -106,7 +116,33 @@ func (r *phaseReconciler) reconcile(ctx context.Context, workspace *tenancyv1alp
 			logger.Info("workspace content is deleted")
 			return reconcileStatusContinue, nil
 		}
+		logger.Info("workspace is ready, checking conditions", "conditions", workspace.Status.Conditions)
+		// if workspace is ready, we check if it suppose to be ready by checking conditions.
+		if checkConditionAndSetPhase(workspace) {
+			return reconcileStatusStopAndRequeue, nil
+		}
 	}
 
 	return reconcileStatusContinue, nil
 }
+
+// checkConditionAndSetPhase checks if the workspace is ready by checking conditions and sets the phase accordingly.
+// It returns true if the phase was changed, false otherwise.
+func checkConditionAndSetPhase(workspace *tenancyv1alpha1.Workspace) bool {
```

Comment:
the name does not really fit as it is not responsible for the initial phases. Maybe call it `updateTerminalConditionsAndPhase`?